privval/file.go

```diff
@@ -198,37 +225,57 @@ func LoadFilePVEmptyState(keyFilePath, stateFilePath string) (*FilePV, error) {
 	return loadFilePV(keyFilePath, stateFilePath, false)
 }
 
+func fileAndContents(path string) (*os.File, []byte, error) {
+	f, err := os.OpenFile(path, os.O_RDWR|os.O_CREATE, 0755)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	blob, err := io.ReadAll(f)
+	if err != nil {
+		f.Close()
+		return nil, nil, err
+	}
+	return f, blob, nil
+}
+
 // If loadState is true, we load from the stateFilePath. Otherwise, we use an empty LastSignState.
-func loadFilePV(keyFilePath, stateFilePath string, loadState bool) (*FilePV, error) {
-	keyJSONBytes, err := os.ReadFile(keyFilePath)
+func loadFilePV(keyFilePath, stateFilePath string, loadState bool) (_ *FilePV, lerr error) {
+	keyFile, keyJSONBytes, err := fileAndContents(keyFilePath)
 	if err != nil {
 		return nil, err
 	}
+	defer func() {
+		if lerr != nil {
```

Comment:
Can you elaborate on what his line is doing? `lerr` is never set, so I'm not sure what condition we're checking for here.